src/ocean/net/ssl/openssl/OpenSsl.d

```diff
@@ -0,0 +1,833 @@
+            connection was closed, or a negative value if an error occurred or
+            action must be taken by the calling process.
+
+    ***************************************************************************/
+
+    public int SSL_read (SSL* ssl, void* buf, int num);
+
+
+    /***************************************************************************
+
+        Sets the verification parameters for an SSL context
+
+        Params:
+            ctx = the SSL context to be set
+            mode = the verification flags to use. For a client, this must be
+                SSL_VERIFY_NONE or SSL_VERIFY_PEER.
+            callback = the verification callback to use when mode is set to
+                SSL_VERIFY_PEER, or null to use the default callback
+
+    ***************************************************************************/
+
+    public void SSL_CTX_set_verify (SSL_CTX* ctx, int mode,
+        int function (int, X509_STORE_CTX*) callback);
+
+
+    /***************************************************************************
+
+        Sets the maxiumum depth for certificate chain verification
+
+        Params:
+            ctx = the SSL context for which the depth should be set
+            depth = The maximum depth to be allowed
+
+    ***************************************************************************/
+
+    public void SSL_CTX_set_verify_depth (SSL_CTX* ctx, int depth);
+
+
+    /***************************************************************************
+
+        Specifies the locations for ctx, at which CA certificates for
+        verification purposes are located. The certificates available via
+        CAfile and CApath are trusted.
+
+        When looking up CA certificates, the OpenSSL library will first search
+        the certificates in CAfile, then those in CApath.
+
+        Params:
+            ctx = the CTX
+            CAfile = pointer to a file of CA certificates in PEM format, or
+                null. The file can containe several CA certificates.
+            CApath = a directory containing CA certificates in PEM format
+
+        Returns:
+            0 if the operation fails, 1 if the operation was successful
+
+    ***************************************************************************/
+
+    public int SSL_CTX_load_verify_locations (SSL_CTX* ctx,
+        Const!(char*) CAfile, Const!(char*) CApath);
+
+
+    /***************************************************************************
+
+        Get the result of peer certficate verification
+
+        Params:
+            ssl = the SSL object which obtained the peer certificate
+
+        Returns:
+            X509_V_OK if the verification succeeded or no peer certificate was
+            presented, ot an error code if the verification failed.
+
+    ***************************************************************************/
+
+    public c_long SSL_get_verify_result (/*const*/ SSL* ssl);
```

Comment:
What's the reason for this not being `Const!(SSL)` and as you have `Const!(char*) CAfile` in other method that accepts parameter as const?

Reply:
I have used `Const` whenever it's ultimately a string, so that it works with string literals. In the case where they are opaque structs it doesn't really make any difference if it is const or not, I think.
Do we have a policy about constness of C bindings?

Reply:
> Do we have a policy about constness of C bindings?

Explicitly ? No AFAIK. But most (all?) binding I've seen just sticked to what C does.
If it's a pointer to constant data, then it has a `const`.